### ISSUE TITLE
Compatibility with Gradle 4.8

### ISSUE THREAD

**Issue by trevjonez** [wupdigital/android-maven-publish#19]
```
java.lang.AbstractMethodError: digital.wup.android_maven_publish.AndroidVariantLibrary$CompileUsage.getGlobalExcludes()Ljava/util/Set;
        at org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication.populateFromComponent(DefaultMavenPublication.java:246)
        at org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication.getPublishableArtifacts(DefaultMavenPublication.java:357)
        at org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication_Decorated.getPublishableArtifacts(Unknown Source)
        at org.gradle.api.publish.maven.tasks.AbstractPublishToMaven$1.call(AbstractPublishToMaven.java:46)
        at org.gradle.api.publish.maven.tasks.AbstractPublishToMaven$1.call(AbstractPublishToMaven.java:43)
        at org.gradle.util.GUtil.uncheckedCall(GUtil.java:458)
        at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:73)
        at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:75)
        at org.gradle.api.internal.tasks.TaskPropertyFileCollection.visitContents(TaskPropertyFileCollection.java:53)
        at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:187)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:89)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:211)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.doGetDependencies(CachingTaskDependencyResolveContext.java:76)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:60)
[...]
```

**Comment by trevjonez** [comment]
Looks like you already got through #18 & #17 today. I was going to take a swing at this one today as I am really wanting to use the new pom dsl they added in gradle 4.8 but it seems you might already be working on the solution? Let me know, I have a handful of libraries I would love to upgrade to use this plugin asap.

**Comment by warnyul** [comment]
Yes, I am working on the solution. I would like to release the new version today.

**Comment by warnyul** [comment]
Soon it will be available in [Gradle Plugin portal](https://plugins.gradle.org/plugin/digital.wup.android-maven-publish)

**Comment by trevjonez** [comment]
looks like the upload from travis failed due to a misconfig some where?

**Comment by warnyul** [comment]
Something wrong with bintrayUpload:

`Could not upload to 'https://api.bintray.com/content/[secure]/[secure]-maven/android-maven-publish/3.6.0/value: digital/wup/value: android-maven-publish/value: 3.6.0/value: android-maven-publish-value: 3.6.0.pom': HTTP/1.1 400 Bad Request [message:Unable to upload files: Maven group, artifact or version defined in the pom file do not match the file path 'value: digital/wup/value: android-maven-publish/value: 3.6.0/value: android-maven-publish-value: 3.6.0.pom']`

**Comment by warnyul** [comment]
I am not so happy. This is an [open issue on bintray plugin](https://github.com/bintray/gradle-bintray-plugin/issues/234). I can't make release today. I will revert this release and reopen the issue.
Until this has been fixed.

**Comment by trevjonez** [comment]
Looks like they shipped gradle-bintray-plugin:1.8.1 today. I was able to deploy this repo locally and upload my project once I had both 3.6.0 of this plugin and 1.8.1 of theirs.

**Comment by warnyul** [comment]
Thansks for your help. I have published 3.6.0 with bintray plugin 1.8.1. It is available now.
